DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -7,201 +7,325 @@
 
 # # imports
 import os
+import time
+import datetime
+import socket
 # # from DIRAC
 from DIRAC import S_OK, S_ERROR, gLogger
 from DIRAC.FrameworkSystem.Client.MonitoringClient import gMonitor
-from DIRAC.DataManagementSystem.Agent.RequestOperations.DMSRequestOperationsBase  import DMSRequestOperationsBase
-from DIRAC.DataManagementSystem.Client.ConsistencyInspector                       import ConsistencyInspector
+from DIRAC.DataManagementSystem.Agent.RequestOperations.DMSRequestOperationsBase import DMSRequestOperationsBase
+from DIRAC.DataManagementSystem.Client.ConsistencyInspector import ConsistencyInspector
+
+from DIRAC.MonitoringSystem.Client.MonitoringReporter import MonitoringReporter
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
 
 ####
-class MoveReplica( DMSRequestOperationsBase ):
+
+
+class MoveReplica(DMSRequestOperationsBase):
   """
   .. class:: MoveReplica
 
   MoveReplica operation handler
   """
 
-  def __init__( self, operation = None, csPath = None ):
+  def __init__(self, operation=None, csPath=None):
     """c'tor
 
     :param self: self reference
     :param ~Operation.Operation operation: Operation instance
     :param str csPath: CS path for this handler
     """
-    super( MoveReplica, self ).__init__( operation, csPath )
-    # # own gMonitor stuff for files
-    gMonitor.registerActivity( "ReplicateAndRegisterAtt", "Replicate and register attempted",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "ReplicateOK", "Replications successful",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "ReplicateFail", "Replications failed",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RegisterOK", "Registrations successful",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RegisterFail", "Registrations failed",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RemoveReplicaAtt", "Replica removals attempted",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RemoveReplicaOK", "Successful replica removals",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RemoveReplicaFail", "Failed replica removals",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
+    super(MoveReplica, self).__init__(operation, csPath)
+
+    # Check whether the ES flag is enabled so we can send the data accordingly.
+    self.rmsMonitoring = Operations().getValue("EnableActivityMonitoring", False)
```

Comment:
This does not reflect if we this flag is True but we set it False in the agent option. But you do not have access to the agent options.

Reply:
I think we won't be able to access the service or agent flag from here as they are not instances of any of the two so that's why I choose to use only Operations here but we can do it for `RequestExecutingAgent` as its an agent instance. I will think about this then too.

Reply:
@zmathe can we pass this flag from RequestExecutingAgent to the `RequestOperations` this way we will be able to keep the flag configurable from both Operations and Agent?

Reply:
Solved this problem.